internal/component/database_observability/mysql/collector/query_sample.go

```diff
@@ -275,3 +271,68 @@ func (c *QuerySample) fetchQuerySamples(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *QuerySample) calculateTimestamp(serverStartTime, timer float64) uint64 {
+	const millisecondsPerSecond float64 = 1e3
+	// timer indicates event timing since server startup.
+	// The timer value is in picoseconds with a column type of bigint unsigned. This value can overflow after about ~213 days.
+	// We need to account for this overflow when calculating the timestamp.
+
+	// Knowing the number of overflows that occurred, we can calculate how much overflow time to compensate.
+	previousOverflows := calculateNumberOfOverflows(c.lastUptime)
+	overflowTime := float64(previousOverflows) * picosecondsToSeconds(float64(math.MaxUint64))
+	// We then add this overflow compensation to the server start time, and also add the timer value (remember this is counted from server start).
+	// The resulting value is the timestamp in seconds at which an event happened.
+	timerSeconds := picosecondsToSeconds(timer)
+	timestampSeconds := serverStartTime + overflowTime + timerSeconds
+
+	return uint64(timestampSeconds * millisecondsPerSecond) // convert to milliseconds
+}
+
+func calculateNumberOfOverflows(uptime float64) int {
+	return int(math.Floor(uptime / picosecondsToSeconds(float64(math.MaxUint64))))
```

Comment:
I see `picosecondsToSeconds(float64(math.MaxUint64))` is used in a couple of places, could it be a const?